src/submit.py

```diff
@@ -2,78 +2,103 @@
 
 import asyncio
 import pathlib
+import re
+import textwrap
 
 import aiofiles
 import click
 
 import src.clock as clock
-from src.settings import AGENT_HOME_DIR, HOOKS, async_cleanup
+import src.settings as settings
 
-_SUBMISSION_PATH = AGENT_HOME_DIR / "submission.txt"
+_SUBMISSION_PATH = settings.AGENT_HOME_DIR / "submission.txt"
 
 
-async def _git_push(repo_dir: pathlib.Path) -> tuple[int, str]:
+async def run_command(command: list[str], cwd: pathlib.Path) -> tuple[int, str]:
     process = await asyncio.subprocess.create_subprocess_exec(
-        "git",
-        "push",
-        cwd=repo_dir,
+        *command,
+        cwd=cwd,
         stdout=asyncio.subprocess.PIPE,
         stderr=asyncio.subprocess.STDOUT,
     )
     stdout, _ = await process.communicate()
-    output = stdout.decode().strip()
     return_code = await process.wait()
-    return return_code, output
+    return return_code, stdout.decode().strip()
+
+
+async def _git_push(repo_dir: pathlib.Path) -> tuple[int, str]:
+    return await run_command(["git", "push"], repo_dir)
 
 
 async def _create_submission_commit(repo_dir: pathlib.Path):
-    # Stash any changes
-    stash_process = await asyncio.subprocess.create_subprocess_exec(
-        "git",
-        "stash",
-        "push",
-        cwd=repo_dir,
-        stdout=asyncio.subprocess.PIPE,
-        stderr=asyncio.subprocess.STDOUT,
+    commands = [
+        ["git", "stash", "push"],
+        ["git", "commit", "--allow-empty", "-m", "SUBMISSION"],
+        ["git", "stash", "pop"],
+    ]
+    for command in commands:
+        await run_command(command, repo_dir)
+
+
+async def git_clone_instructions(repo_dir: pathlib.Path):
+    _, ip_address = await run_command(["hostname", "-I"], cwd=repo_dir)
```

Comment:
this could also be extracted from the `ssh -G` command, but I for some reason trust this one more...